os/include/tinyara/mm/mm.h

```diff
@@ -351,6 +351,18 @@ extern struct heapinfo_group_info_s group_info[HEAPINFO_THREAD_NUM];
 #endif
 
 #if CONFIG_KMM_NHEAPS > 1
+struct heapinfo_total_info_s {
+	int total_heap_size;
+	int cur_free;
+	int largest_free_size;
+	int cur_dead_thread;
+	int sum_of_stacks;
+	int sum_of_heaps;
+	int cur_alloc_size;
+	int peak_alloc_size;
+};
+typedef struct heapinfo_total_info_s heapinfo_total_info_t;
```

Comment:
1. It should changed to um.. like a public structure? parser is only caller who calling this structure, you can move it to there to use it as a private structure.
2. Some of values in this structure had better change its type to size_t instead of int to maintain more huge number of size & prevent error.

Reply:
1. In mm_heapinfo_utils.c, there is also a variable, `heapinfo_total_info_t total_info;`.
For using here, it should be declared in header.
2. Okay. I will check and change the type with another PR. Thanks.

Reply:
> 1. In mm_heapinfo_utils.c, there is also a variable, `heapinfo_total_info_t total_info;`.
>    For using here, it should be declared in header.
Actually I couldn't see it in heapinfo_utils.c is it newly added?
Alright, anyway it still can be hided but not major structure, so let it be.

Reply:
then you can remove below extern, I think